### PR TITLE
Implement to join battle for friend

### LINKED ISSUE #209: Implement to join battle for friend
Screenshot_20230511_233511
Implement this button

### PR DESCRIPTION
**Pull request by 1-alex98**
Fixes #209

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (1-alex98) [inline .vscode/launch.json:1]: I can also take that back but I find that pretty handy for debugging out of vsc.
And why not check that in...

File: package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   "main": "./out/main/index.js",
   "scripts": {
     "dev": "electron-vite",
+    "debug": "electron-vite --remote-debugging-port=9223",
```
Comment (1-alex98) [inline package.json:14]: used by launch.json

File: src/renderer/api/session.ts [hunk 3]
```diff
@@ -184,4 +184,33 @@ export class SessionAPI {
         await nextTick();
         return this.fetchUserById(userId);
     }
+
+    public async updateBattleList() {
```
Comment (1-alex98) [inline src/renderer/api/session.ts:188]: Moved here so it can be reused

File: src/renderer/components/navbar/Friend.vue [hunk 4]
```diff
@@ -117,8 +118,18 @@ function sendMessage() {
 }
 
 async function joinBattle() {
-    // TODO: need to know about the battle before attempting to join it because it may be passworded
-    // await attemptJoinBattle();
+    const battleIdToJoin = props.user.battleStatus.battleId;
+    await api.session.updateBattleList();
```
Comment (1-alex98) [inline src/renderer/components/navbar/Friend.vue:122]: Make sure the battle list is acctaully up to date

**Review by Jazcash — APPROVED** [review]
Sensible and clean - thanks for the PR!

**Pull request merged**